database/maintenance.go

```diff
@@ -139,7 +139,7 @@ func duplicateControllers() (all []*Controller) {
 	controllersLock.RLock()
 	defer controllersLock.RUnlock()
 
-	all = make([]*Controller, len(controllers))
+	all = make([]*Controller, 0, len(controllers))
 	for _, c := range controllers {
 		all = append(all, c)
 	}
```

Comment:
```suggestion
	all = make([]*Controller, len(controllers))
	copy(all, controllers)
```

Reply:
Nevermind those comment. Since `controllers` is a map that's not going to work.
:facepalm: